pkg/client/conditions.go

```diff
@@ -22,11 +22,28 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait"
 )
 
+type ConditionsNamespacer interface {
+	Conditions(namespace string) ConditionsInterface
+}
+
+type ConditionsInterface interface {
+	ControllerHasDesiredReplicas(*api.ReplicationController) wait.ConditionFunc
+}
+
+type conditions struct {
+	r  Interface
+	ns string
+}
+
+func newConditions(c Interface, namespace string) *conditions {
+	return &conditions{c, namespace}
+}
+
 // ControllerHasDesiredReplicas returns a condition that will be true iff the desired replica count
 // for a controller's ReplicaSelector equals the Replicas count.
-func (c *Client) ControllerHasDesiredReplicas(controller api.ReplicationController) wait.ConditionFunc {
+func (c *conditions) ControllerHasDesiredReplicas(controller *api.ReplicationController) wait.ConditionFunc {
 	return func() (bool, error) {
-		pods, err := c.Pods(controller.Namespace).List(labels.Set(controller.Spec.Selector).AsSelector())
+		pods, err := c.r.Pods(c.ns).List(labels.Set(controller.Spec.Selector).AsSelector())
```

Comment:
This won't return the correct answer. You need to do the same thing as kubectl describe rc:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/describe.go

Don't count the pods. Use rc.Status.Replicas.